domain/src/main/java/life/qbic/usermanagement/registration/Registration.java

```diff
@@ -0,0 +1,51 @@
+package life.qbic.usermanagement.registration;
+
+import life.qbic.usermanagement.User;
+import life.qbic.usermanagement.repository.UserDataStorage;
+import life.qbic.usermanagement.repository.UserRepository;
+
+/**
+ * <b>User Registration use case</b>
+ * <p>
+ * Tries to register a new user and create a user account.
+ * <p>
+ * In case a user with the provided email already exists, the registration will fail and calls the
+ * failure output method.
+ *
+ * @since 1.5.0
+ */
+public class Registration implements RegisterUserInput {
+
+  private RegisterUserOutput registerUserOutput;
+
+  private final UserRepository userRepository;
+
+  public Registration(UserRepository userRepository) {
+    this.userRepository = userRepository;
+  }
+
+  public void setRegisterUserOutput(RegisterUserOutput registerUserOutput) {
+    this.registerUserOutput = registerUserOutput;
+  }
+
+  @Override
+  public void register(User user) {
+    userRepository.findByEmail(user.getEmail())
+        .ifPresentOrElse(
+            u -> registerUserOutput.onFailure("User with email address already exists."),
+            () -> {
+              user.setEmailConfirmed(false);
+              userRepository.addUser(user);
+              registerUserOutput.onSuccess();
```

Comment:
you might want to make sure, this was set properyl? Now the output could be null

Reply:
yes, tbh I dont like how it is currently set up to break the cyclic dependency. A use case without an output is just wrong by design. 

I dont have a better idea though how to solve it.